arghandling.h

```diff
@@ -1,5 +1,8 @@
 /* Taken from emacs */
 #define ARRAYELTS(arr) (sizeof (arr) / sizeof (arr)[0])
 
-extern "C" bool
+#ifdef __cplusplus
```

Comment:
This looks good, although most compilers don't need it, and will simply ignore the `extern "C"` thing.

Reply:
but for c++ compiler it will mangle the name and throw a link error